"Links with identical accessible names have equivalent purpose" (b20e66): Prepare for TF review (#1463)
act-rules/act-rules.github.io: _rules/links-identical-name-equivalent-purpose-b20e66.md

#### Passed Example 2

> A set of two HTML `a` elements that both resolve to [same resource][] after instant redirect. The redirect means the URLs look different but load the same page.

Reviewer: ```suggestion
These two HTML `a` elements resolve to the [same resource][] after instant redirect. The redirect means that the URLs look different but load the same page.
```

#### Passed Example 3

> These two links resolve to resources that are not the same (different URLs), but the resources are completely identical, in this case the contact details being the same in 2 locations, thus serving the same purpose.

Reviewer 2: ```suggestion
These two links resolve to resources that are not the same (different URLs), but the resources are completely identical, in this case the contact details being the same in two locations, thus serving the same purpose.
```

#### Passed Example 6

> These two links go to pages that contain different amounts of information and/or differently worded information, but fulfill the same purpose in relation to the link. In the example, the call us details are the same format but in different locations.

Reviewer: ```suggestion
These two links go to pages that contain different amounts of information and/or differently worded information, but fulfill the same purpose in relation to the link. In the example, the phone number is in the same format but in different locations.
```

#### Passed Example 9

> A set of two SVG `a` elements have the same [accessible name][] and link to the [same resource][]. The example is one SVG area but with 2 formats of links. One uses an image and the other uses text.

Reviewer: ```suggestion
These two SVG `a` elements have the same [accessible name][] and link to the [same resource][]. The example is one SVG area but with 2 formats of links. One uses an image and the other uses text.
```

#### Passed Example 11

> All three links have the same [accessible name][]. The second link ("from the light") is only part of the [light tree][]. When the [shadow tree][] is attached to `host` and flattened, this link is overwritten and therefore not part of the [flat tree][]. Hence, only the first and third link are considered by this rule and they both point to the [same resource][]. The second link is not part of the page rendered to the user.

Reviewer: ```suggestion
All three links have the same [accessible name][]. The second link ("from the light") is only part of the [light tree][]. When the [shadow tree][] is attached to `host` and flattened, this link is overwritten and therefore not part of the [flat tree][] (and thus not rendered). Hence, only the first and third link are considered by this rule and they both point to the [same resource][].
```

#### Failed Example 1

> Both links have the same [accessible name][] but go to different resources. They open completely different pages.

Reviewer: ```suggestion
These two links have the same [accessible name][] but go to different resources. They open completely different pages.
```

#### Failed Example 5

> A set of two SVG `a` elements that have the same [accessible name][] but link to different resources. They open completely different pages.

Reviewer: ```suggestion
These two SVG `a` elements have the same [accessible name][] but link to different resources. They open completely different pages.
```

#### Failed Example 6

> Both links resolve to [same resource][] after redirect, but the redirect is not instant. The user will notice the redirect due to a pause.

Reviewer: ```suggestion
These two links resolve to [same resource][] after redirect, but the redirect is not instant. The user will notice the redirect due to a pause.
```

#### Inapplicable Example 1

> These `a` and `area` elements have no `href` attribute. These are not valid links.

Reviewer: ```suggestion
These `a` and `area` elements have no `href` attribute. Thus, they do not have a role of `link`.
```

Author: I'm not sure on this as the 'a' and 'area' are the role of link, but the 'href' is where it goes.

Reviewer: According to the mappings: https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings (direct links to the elements don't work well… you need to scroll… 😞 ), the `a` and `area` elements have no role when there is no `href` attribute.
So, in that case, there is no element with a role of `link` (or inheriting from `link`), hence the rule is not applicable. I say, "role of `link`" is the technical term for "valid link" 😀

Reviewer: "not work as expected" is a fairly strong statement, given that there are valid cases for `a` without `href`: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element (see the second paragraph after the green box + the first example). So, it might be a deliberate choice of the author to omit the `href` attribute, and the element would work exactly as expected.

(+ see the example at https://html.spec.whatwg.org/multipage/image-maps.html#image-maps for an `area` without `href` where the lack of `href` is exactly what is intended, namely to make the "hole" in the red square not clickable)

Author: So in plain text how would you define a link without a role?

Reviewer: Well, that's the point. It's not a link.
It's an `a`/`area` element without an `href` attribute and therefore not a link (role has little to do here, it's not a link for ARIA (no role), but it's not a link for HTML in the first place (not clickable, …)

Maybe "Thus they are not links and do not have a role of `link`"?

Author: Yes - that would work 👍

Author: @Jym77, can I update the last suggestion to the change to commit? Or is it something for you to change?

Reviewer: Please do it.
I normally prefer to let one person handle all changes on a PR, so that there is clear responsibility (not in term of "who to blame?", but rather in avoiding the "who should do that?" and streamlining the process).

> These `a` and `area` elements have no `href` attribute. Thus they are not links and do not have a role of link

Reviewer: ```suggestion
These `a` and `area` elements have no `href` attribute. Thus they are not links and do not have a role of `link`.
```

#### Inapplicable Example 3

> These `span` elements do not have a [semantic role][] of link. They are not valid links.

Reviewer: ```suggestion
These `span` elements do not have a [semantic role][] of `link`. They are not valid links.
```
